drivers/i2c/i2c_cc13xx_cc26xx.c

```diff
@@ -206,7 +206,8 @@ static int i2c_cc13xx_cc26xx_transfer(struct device *dev, struct i2c_msg *msgs,
 
 	k_sem_take(&get_dev_data(dev)->lock, K_FOREVER);
 
-#ifdef CONFIG_SYS_POWER_MANAGEMENT
+#if defined(CONFIG_SYS_POWER_MANAGEMENT) && \
+	defined(CONFIG_SYS_POWER_SLEEP_STATES)
```

Comment:
Should this be:

    defined(CONFIG_HAS_SYS_POWER_STATE_SLEEP_2)

Because at the moment each sleep level is independently controlled (!!).  Happens the board currently selects that, but if somebody changes it this would break again.

Reply:
Thanks. I think the condition should be removed from `subsys/power/policy/policy_residency_cc13x2_cc26x2.c`. In the [review](https://github.com/zephyrproject-rtos/zephyr/pull/21465#pullrequestreview-349642793) of #21465, we made the decision that it'd be ok to assume `CONFIG_HAS_SYS_POWER_STATE_SLEEP_1` and `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` are set since they are set at SoC level for this platform. It'd be good to stay consistent.

Meanwhile, `defined(CONFIG_HAS_SYS_POWER_STATE_SLEEP_2)` would not fix the problem for the case when `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` is defined but `CONFIG_SYS_POWER_SLEEP_STATES` isn't. The build error would still occur.

Reply:
>  would not fix the problem for the case when `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` is defined but `CONFIG_SYS_POWER_SLEEP_STATES` isn't

I would hope that combination would be rejected by Kconfig, but don't think it's worth checking.